frontend/src/features/public-form/utils/inputValidation.ts

```diff
@@ -0,0 +1,94 @@
+import {
```

Comment:
what's the value of this validation layer over what we're doing in `validateResponses` in #3600? it seems rather complicated to validate the answers in the form, then validate them again on submission, then reshape them, then validate them again in `validateResponses`.

I think `EncryptionService` made sense as a final check before end-to-end encryption, but personally I'm not sure this validation adds much value - we're just re-validating outputs which are already in our control.

Reply:
the original intention of this validation layer is meant to act as a typeguard so the correct types are inferred after we pass it through this layer. Since a typeguard is going to be written, I thought we might as well correctly validate it.

But now that you say that, we can also create a proper union type that maps `BasicField` to their react-hook-form `FieldValues`. I will experiment to see if I can drop inputValidation layer completely.

Reply:
I think the only way is via casting, do you think that's ok?

```ts
/// inputTransformation.ts
export const transformInputsToOutputs = (
  field: FormFieldDto,
  input: unknown,
): FieldResponse | undefined => {
  switch (field.fieldType) {
    case BasicField.Section:
      return transformToSectionOutput(field)
    case BasicField.Checkbox:
      return transformToCheckboxOutput(field, input as CheckboxFieldValues) // <--
...
}
```

Then we can remove the validation layer. But I feel like it would not be that big of a deal to pass it through the layer as well.

Reply:
would it help if we pass generic types as per this comment? https://github.com/opengovsg/FormSG/pull/3623#discussion_r841678908

rn it seems like we are doing validation 4 times on the frontend:
1. `react-hook-form` validators
2. validate every single field for form logic upon every single change to the form
3. validate every single field upon form submission
4. validate every single response after it gets transformed from a field

it seems like step 2 and 3 are redundant, and (not sure about the magnitude but) might cause a performance hit on large forms bc we'll be validating every single form field for every change (e.g. character typed).

Reply:
Done. Using type casting instead. 5e1d5dd